Makefile

```diff
@@ -31,3 +31,45 @@ deep-copy:
 crds:
 	operator-sdk generate crds
 	mv ./deploy/crds/* ./charts/piraeus/crds
+
+release:
+	# check that VERSION is set
+	@if [ -z "$(VERSION)" ]; then \
+		echo "Usage: make prepare-release VERSION=<version>" >&2 ; \
+		exit 1 ; \
+	fi
+	# check that version has expected format
+	@if ! echo -e "$(VERSION)" | grep -cP '^\d+\.\d+\.\d+$$' > /dev/null ; then \
+		echo "version format: <maj>.<min>.<patch>" >&2 ; \
+		exit 1 ; \
+	fi
+	# check that version does not exist
+	@if git rev-parse "v$(VERSION)" >/dev/null 2>&1 ; then \
+		echo "version v$(VERSION) already exists" >&2 ; \
+		exit 1 ; \
+	fi
+	# check that working tree is clean
+	@if ! git diff-index --quiet HEAD -- ; then \
+			echo "Refusing to create release from dirty repository" >&2 ; \
```

Comment:
does not really matter, but compared to the others this has one additional indent level.